features/admin.applications.v1/components/edit-application.tsx

```diff
@@ -491,21 +491,34 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
 
         if(isEmpty(window.location.hash)){
 
-            if(urlSearchParams.get(ApplicationManagementConstants.APP_STATE_STRONG_AUTH_PARAM_KEY) !==
+            if(urlSearchParams.get(ApplicationManagementConstants.APP_STATE_STRONG_AUTH_PARAM_KEY) ===
                 ApplicationManagementConstants.APP_STATE_STRONG_AUTH_PARAM_VALUE) {
-                handleDefaultTabIndexChange(defaultTabIndex);
+                // When application selection is done through the strong authentication flow.
+                const signInMethodtabIndex: number = renderedTabPanes?.findIndex(
+                    (element: {"componentId": string}) =>
+                        element.componentId === ApplicationManagementConstants.SIGN_IN_METHOD_TAB_URL_FRAG
+                );
+
+                if (signInMethodtabIndex !== -1) {
+                    handleActiveTabIndexChange(signInMethodtabIndex);
+                }
 
                 return;
-            }
+            } else if (urlSearchParams.get(ApplicationManagementConstants.IS_PROTOCOL) ===
+                ApplicationManagementConstants.APP_STATE_STRONG_AUTH_PARAM_VALUE) {
+                const protocolTabIndex: number = renderedTabPanes?.findIndex(
+                    (element: {"componentId": string}) =>
+                        element.componentId === ApplicationManagementConstants.PROTOCOL_TAB_URL_FRAG
+                );
```

Comment:
This logic is not going to be triggered on URL param change, as the dependencies of `useEffect` doesn't contain urlSearchParams. Therefore, let's add urlSearchParams to the dependency array to make this work.

Reply:
Added with 66db63647fd006555a02f3781c73a0c098394830